tools/crm_resource_ban.c

```diff
@@ -395,7 +384,10 @@ build_clear_xpath_string(GString *buf, const xmlNode *constraint_node,
         } else if (promoted_role_only) {
             rsc_role_substr = g_string_sized_new(64);
             pcmk__g_strcat(rsc_role_substr,
-                           "@" XML_RULE_ATTR_ROLE "='", promoted_role_name(),
+                           "@" XML_RULE_ATTR_ROLE "='",
```

Comment:
We don't `pcmk__g_strcat()` now. All of this is a single string. Should be something like:
```
            rsc_role_substr = g_string_sized_new(64);
            g_string_append(rsc_role_substr,
                            "@" XML_RULE_ATTR_ROLE "='" PCMK__ROLE_PROMOTED "' "
                            "or @" XML_RULE_ATTR_ROLE 
                                "='" PCMK__ROLE_PROMOTED_LEGACY "'");
```